BST.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BST
+{
+    class BST
+    {
+
+        public Node root;
+
+
+        public BST()
+        {
+            root = null;
+        }
+
+        public void insert(int key)
+        {
+            root = insertRec(root, key);
+        }
+
+        public Node insertRec(Node root, int key)
+        {
+
+            if (root == null)
+            {
+                root = new Node(key);
+                return root;
+            }
+
+            if (key < root.key)
+                root.left = insertRec(root.left, key);
+            else if (key > root.key)
+                root.right = insertRec(root.right, key);
+
+            return root;
+        }
+
+        public void inorder()
+        {
+            inorderRec(root);
+        }
+
+        public void inorderRec(Node root)
+        {
+            if (root != null)
+            {
+                inorderRec(root.left);
+                Console.WriteLine(root.key);
+                inorderRec(root.right);
+            }
+        }
+
+        public void deleteBinarySearchTree()
+        {
+            this.root = null;
+        }
+
+        public static Node minimumKey(Node curr)
+        {
+            while (curr.left != null)
+            {
+                curr = curr.left;
+            }
+            return curr;
+        }
+
+        public void updateNode(Node root , int oldKey, int newKey) {
+        
+
+            this.deleteNode(root, oldKey);
+
+            this.insert(newKey);
+
+            Console.WriteLine("Tree has been updated");
+
+
+        }
+
+        public void traverseRight(Node root)

```

Comment:
TraverseRight() method is not implemented yet. You shouldn't leave empty methods.